docker/client.py

```diff
@@ -239,7 +241,7 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
+        if six.PY3 and hasattr(socket, "_sock"):
```

Comment:
Fair enough, The problem is that in Python3 _sock still exists even if ssl isn't used, but if we use _sock in that condition then it breaks. Which means we can only use _sock if it's using ssl.